seclea_ai/internal/file_processor.py

```diff
@@ -229,3 +569,45 @@ def _send_model_state(self, save_path, training_run_pk: int, sequence_num: int,
                 f"Response Status code {res.status}, expected: 201. \n f'There was some issue uploading a model state: {res.text()}' - {res.reason} - {res.text}"
             )
         return res
+
+    @staticmethod
+    def _ensure_required_metadata(metadata: Dict, defaults_spec: Dict) -> Dict:
+        """
+        Ensures that required metadata that can be specified by the user are filled.
+        @param metadata: The metadata dict
+        @param defaults_spec:
+        @return: metadata
+        """
+        for required_key, default in defaults_spec.items():
+            try:
+                if metadata[required_key] is None:
+                    metadata[required_key] = default
+            except KeyError:
+                metadata[required_key] = default
+        return metadata
+
+    @staticmethod
+    def _add_required_metadata(metadata: Dict, required_spec: Dict) -> Dict:
+        """
+        Adds required - non user specified fields to the metadata
+        @param metadata: The metadata dict
+        @param required_spec:
+        @return: metadata
+        """
+        for required_key, default in required_spec.items():
+            metadata[required_key] = default
+        return metadata
+
+    @staticmethod
+    def _get_framework(model) -> ModelManagers:
+        module = model.__class__.__module__
+        # order is important as xgboost and lightgbm contain sklearn compliant packages.
+        # TODO check if we can treat them as sklearn but for now we avoid that issue by doing sklearn last.
+        if "xgboost" in module:
+            return ModelManagers.XGBOOST
+        elif "lightgbm" in module:
+            return ModelManagers.LIGHTGBM
```

Comment:
All this validation code should stay in seclea_ai. We should only send data to the saving/sending infrastructure when it has been fully validated.

Reply:
Ok will move it

Reply:
Thanks

Reply:
Resolved this.